lib/Runtime/ByteCode/ByteCodeEmitter.cpp

```diff
@@ -1578,15 +1578,19 @@ void ByteCodeGenerator::EmitScopeObjectInit(FuncInfo *funcInfo)
                     Assert(sym->GetScopeSlot() != Js::Constants::NoProperty && sym->GetScopeSlot() > slot);
                     propIds->elements[slot] = Js::Constants::NoProperty;
                 }
-                slot++;
             }
+            else
```

Comment:
Either it will be var sym or it will be pattern. Putting explicit assert is for pattern is redundant. GetFormalNext() already has assert.